File: C_Cpp/swap/src/swap.cpp
//============================================================================
// Name        : swap.cpp
// Author      : 
// Version     :
// Copyright   : Your copyright notice
// Description : Hello World in C++, Ansi-style
//============================================================================

#include <iostream>
using namespace std;

void swap(int& x, int& y){
	int temp = x;
	x = y;
	y = temp;
}

int main() {
	int a = 6,b = 10;
	swap(a,b);
	cout << a << " " << b << endl; // prints !!!Hello World!!!
	return 0;
}

File: Java/Info_3_2015_M1_Java_swap_java/src/swap_java/Swap.java
package swap_java;

class A {
	int x;
	static void swap(A x, A y){
		A temp = x;
		x = y;
		y = temp;
	}
	
	static int f(int x){
		int z = x + 10;
		//int h = f(z) +100;
		return f(z);
	}
	
}

public class Swap {
	public static void main(String[] args) {
		A.f(1000);
		A a = new A();
		a.x = 10;
		A b = new A();
		b.x = 20;
		System.out.println(a.x + " " + b.x);
		A.swap(a, b);
		System.out.println(a.x + " " + b.x);
		
	}

}
